change-notes/1.23/analysis-javascript.md

```diff
@@ -16,6 +16,7 @@
 |---------------------------------------------------------------------------|-------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | Unused index variable (`js/unused-index-variable`)                        | correctness                                                       | Highlights loops that iterate over an array, but do not use the index variable to access array elements, indicating a possible typo or logic error. Results are shown on LGTM by default. |
 | Loop bound injection (`js/loop-bound-injection`)                          | security, external/cwe/cwe-834                                      | Highlights loops where a user-controlled object with an arbitrary .length value can trick the server to loop indefinitely. Results are not shown on LGTM by default. |
+| Unreachable method overloads (`js/unreachable-method-overloads`)          | correctness, typescript                                           | Highlights method overloads that are impossible to use from client code. Results are shown on LGTM by default. |
```

Comment:
Could you also add this query to one of the legacy suites?

Reply:
In `javascript/config/suites/javascript`?

Got a specific one in mind? Otherwise I might add it to `maintainability-more`.

Reply:
Yes. `maintainability-more` is fine, or it could even go into `maintainability-core`. But it doesn't matter terribly, since those suites are rarely used these days.